web/ui/react-app/src/pages/graph/HistogramChart.tsx

```diff
@@ -5,19 +5,86 @@ import { bucketRangeString } from './DataTable';
 
 type ScaleType = 'linear' | 'exponential';
 
+type closestToZeroType = {
+  closest: number;
+  closestIdx: number;
+};
+
 const HistogramChart: FC<{ histogram: Histogram; index: number; scale: ScaleType }> = ({ index, histogram, scale }) => {
   const { buckets } = histogram;
+  const formatter = Intl.NumberFormat('en', { notation: 'compact' });
+
   const rangeMax = buckets ? parseFloat(buckets[buckets.length - 1][2]) : 0;
+  const rangeMin = buckets ? parseFloat(buckets[0][1]) : 0;
+
+  const expBucketWidth = buckets
+    ? Math.abs(
+        Math.log(Math.abs(parseFloat(buckets[buckets.length - 1][2]))) -
+          Math.log(Math.abs(parseFloat(buckets[buckets.length - 1][1])))
+      )
+    : 0; //bw
+
   const countMax = buckets ? buckets.map((b) => parseFloat(b[3])).reduce((a, b) => Math.max(a, b)) : 0;
-  const formatter = Intl.NumberFormat('en', { notation: 'compact' });
-  const positiveBuckets = buckets?.filter((b) => parseFloat(b[1]) >= 0); // we only want to show buckets with range >= 0
-  const xLabelTicks = scale === 'linear' ? [0.25, 0.5, 0.75, 1] : [1];
+
+  // The count of a histogram bucket is represented by its area rather than its height. This means it considers
+  // both the count and the width (range) of the bucket. For this, we can set the height of the bucket proportional
+  // to its frequency density (fd). The fd is the count of the bucket divided by the width of the bucket.
+
+  // Frequency density histograms are necessary when the bucekts are of unequal width. If the buckets are collected in
+  // intervals of equal width, then there is no difference between frequency and frequency density histograms.
+  const fds = buckets
+    ? buckets.map((b) => {
+        return parseFloat(b[3]) / (parseFloat(b[2]) - parseFloat(b[1]));
+      })
+    : [];
+  const fdMax = fds.reduce((a, b) => Math.max(a, b));
+  const closestToZero = buckets ? findClosestToZero(buckets.map((b) => parseFloat(b[1]))) : { closest: 0, closestIdx: 0 };
+
+  const zeroAxisLeft =
+    scale === 'linear'
+      ? ((0 - rangeMin) / (rangeMax - rangeMin)) * 100 + '%'
+      : (closestToZero.closestIdx / (buckets ? buckets.length : 1)) * 100 + '%';
+
+  const maxPositive = buckets ? parseFloat(buckets[buckets.length - 1][2]) : 0;
+  const minPositive = buckets ? parseFloat(buckets[closestToZero.closestIdx + 1][1]) : 0;
+  const maxNegative = buckets ? parseFloat(buckets[closestToZero.closestIdx - 1][2]) : 0;
+  const minNegative = buckets ? parseFloat(buckets[0][1]) : 0;
+  const startNegative = buckets ? Math.log(Math.abs(minNegative)) : 0; //start_neg
+  const endNegative = buckets ? Math.log(Math.abs(maxNegative)) : 0; //end_neg
+  const startPositive = buckets ? Math.log(minPositive) : 0; //start_pos
+  const endPositive = buckets ? Math.log(maxPositive) : 0; //end_pos
+
+  const widthNegative = startNegative - endNegative; //width_neg
```

Comment:
With the changes above, this becomes again the more intuitive:

```suggestion
  const widthNegative = endNegative - startNegative; //width_neg
```